src/ImageRunModal.scss

```diff
@@ -27,10 +27,19 @@
     }
 }
 
+// Ensure the width fits within the screen boundaries (with padding on the sides)
+.pf-c-select__menu {
+    // 3xl is the left+right padding for an iPhone SE;
+    // this works on other screen sizes as well
+    max-width: calc(100vw - var(--pf-global--spacer--3xl));
+}
+
 // Make sure the footer is visible with more then 5 results.
 .pf-c-select__menu-list {
-    max-height: 20rem;
-    overflow-y: scroll;
+    // 35% viewport height is for 1280x720;
+    // since it picks the min of the two, it works everywhere
+    max-height: min(20rem, 35vh);
+    overflow: hidden scroll;
 }
```

Comment:
I don't like how these apply to PF directly without being custom classed (like either existing _under_ or in _addition to_ or _just using_  a Cockpit or page specific class).

That said, These are probably generic enough fixes that we might want to actually consider these as PatternFly fixes, so they might need to be overrides and pushed upstream to PatternFly.

Reply:
(Yes, I modified and added this SCSS myself :grin:, but as the menu is popping up outside of the modal, we can't really subclass it, as far as I can tell. But it is really probably generic enough for an upstream fix.)